internal_worker_base.go

```diff
@@ -127,8 +126,8 @@ func newBaseWorker(options baseWorkerOptions, logger *zap.Logger, metricsScope t
 	return &baseWorker{
 		options:         options,
 		shutdownCh:      make(chan struct{}),
-		pollLimiter:     rate.NewLimiter(rate.Every(time.Millisecond*100), 100),
-		taskLimiter:     rate.NewLimiter(rate.Every(options.maxTaskRateRefreshDuration), options.maxTaskRate),
+		pollLimiter:     rate.NewLimiter(1000, 1),
+		taskLimiter:     rate.NewLimiter(rate.Limit(options.maxTaskPerSecond), 1),
```

Comment:
Do we really need two separate rate limiters? This could cause issues of task sitting in the buffer and getting timedout.

Reply:
https://github.com/uber-go/cadence-client/issues/231